File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        vector<vector<int>> result;
        priority_queue<pair<int,vector<int>>> pq;

        for(int i=0; i<points.size(); i++){
            int distance = points[i][0]*points[i][0] + points[i][1]*points[i][1];

            pq.push({distance,points[i]});

            if(pq.size() > k){
                pq.pop();
            }
        }

        while(!pq.empty()){
            result.push_back(pq.top().second);
            pq.pop();
        }

        return result;
    }
};

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        result = []
        heap = []

        for i in range(len(points)):
            distance = points[i][0] * points[i][0] + points[i][1] * points[i][1]

            heapq.heappush(heap, (-distance, points[i]))

            if len(heap) > k:
                heapq.heappop(heap)

        while heap:
            result.append(heapq.heappop(heap)[1])

        return result